docs/requests/authentication.md

```diff
@@ -46,3 +46,22 @@ Also keep in mind that there are some limitations with the current implementatio
 * there is no caching so each request will result in an additional round-trip (or two in case of proxy and server)
 * authorizationInfo is not supported
 * scalajs supports only md5 algorithm
+
+## OAuth2
+
+You can use sttp with OAuth2. Looking at the [OAuth2 protocol flow](https://tools.ietf.org/html/rfc6749#section-1.2), sttp might be helpful in the second and third step of the process:
+
+1. (A)/(B) - Your UI needs to enable the user to authenticate. Your application will then receive a callback from the authentication server, which will include an authentication code.
+
+2. (C)/(D) - You need to send a request to the authorisation server, passing in the authentication code from step 1. You'll receive an access token in response (and optionally a refresh token). For example, assuming that you're using github as your authentication server, that you've taken the values of `clientId` an `clientSecret` from the github settings, and that you have received `authCode` in step 1 above:
```

Comment:
`clientId` an `clientSecret` => `clientId` and `clientSecret` ?
"github settings" are mentioned here, maybe on short sentence above that we are giving GH based example?

Reply:
That's what I wanted to achieve by starting this sentence with "For example, assuming that you're using github as your authentication server" - but maybe it's still not clear enough 🤔

Reply:
but maybe I can say that if you're using a different auth server, you'll gave a different url and so on

Reply:
I'm sorry, it's good.
You use snake case in the code and camel case above. AFAIR RFC was using snake, so maybe we could change here as well?

Reply:
I'm only using snake case where I want circe to work automatically, i.e. when parsing the response from the auth server as `TokenResponse`